server/middleware/ensureAppointmentExists.js

```diff
@@ -0,0 +1,26 @@
+const dbAppointment = require("../db/models/Appointment");
+
+const ensureAppointmentExists = async (req, res, next) => {
+  // Read the appointmentId from url parameters and perform a query in the database
+  // If the appointment exists, proceed
+  // Otherwise, send a 400 response
+  //
+  // URL must contain:  `/:appointmentId`
+  // Example:
+  // `/api/appointment/:appointmentId`
+  // `/api/appointment/cancel/:appointmentId`
+  // `/api/appointment/reschedule/:appointmentId`
+
+  const { appointmentId } = req.params;
+  const result = await dbAppointment.findOne({ _id: appointmentId });
```

Comment:
If you're finding the appointment in the middleware, it doesn't make sense to find it again in the actual controller. You could just add this to the request before moving on

Reply:
That's true. I'll make the edit!